lightning/pkg/importer/precheck_impl.go

```diff
@@ -1433,3 +1435,73 @@ func hasDefault(col *model.ColumnInfo) bool {
 	return col.DefaultIsExpr || col.DefaultValue != nil || !mysql.HasNotNullFlag(col.GetFlag()) ||
 		col.IsGenerated() || mysql.HasAutoIncrementFlag(col.GetFlag())
 }
+
+type pdTiDBFromSameClusterCheckItem struct {
+	db            *sql.DB
+	pdAddrsGetter func(context.Context) []string
+}
+
+// NewPDTiDBFromSameClusterCheckItem creates a new pdTiDBFromSameClusterCheckItem.
+func NewPDTiDBFromSameClusterCheckItem(
+	db *sql.DB,
+	pdAddrsGetter func(context.Context) []string,
+) precheck.Checker {
+	return &pdTiDBFromSameClusterCheckItem{
+		db:            db,
+		pdAddrsGetter: pdAddrsGetter,
+	}
+}
+
+func (i *pdTiDBFromSameClusterCheckItem) Check(ctx context.Context) (*precheck.CheckResult, error) {
+	theResult := &precheck.CheckResult{
+		Item:     i.GetCheckItemID(),
+		Severity: precheck.Critical,
+		Passed:   true,
+		Message:  "PD and TiDB in configuration are from the same cluster",
+	}
+
+	pdAddrs := i.pdAddrsGetter(ctx)
+	pdAddrsMap := make(map[string]struct{}, len(pdAddrs))
+	for _, addrURL := range pdAddrs {
+		u, err2 := url.Parse(addrURL)
+		if err2 != nil {
+			return nil, errors.Trace(err2)
+		}
+		pdAddrsMap[u.Host] = struct{}{}
+	}
+
+	pdAddrsFromTiDB := make([]string, 0, len(pdAddrs))
+	rows, err := i.db.QueryContext(ctx, "SELECT STATUS_ADDRESS FROM INFORMATION_SCHEMA.CLUSTER_INFO WHERE TYPE = 'pd'")
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var addr string
+		err = rows.Scan(&addr)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		// if intersection is not empty, we can say URLs from TiDB and PD are from the same cluster
```

Comment:
please comment that there will be false positive if user configs a LB address

Reply:
if it fails to get, will return user configured address
```
		pdAddrsGetter = func(ctx context.Context) []string {
			leaderInfo, err := pdHTTPCli.GetLeader(ctx)
			if err != nil {
				return []string{cfg.TiDB.PdAddr}
			}
			addrs := leaderInfo.GetClientUrls()
			if len(addrs) == 0 {
				return []string{cfg.TiDB.PdAddr}
			}
```

Reply:
https://github.com/pingcap/tidb/pull/57709/commits/222a2ed70abcc4ef02b55368f9e4dfc4577caf23